src/test/groovy/graphql/nadel/RemovedFieldsTest.groovy

```diff
@@ -849,4 +850,373 @@ class RemovedFieldsTest extends StrategyTestHelper {
+                "Issues",
+                issueSchema,
+                "UserService",
+                userServiceSchema,
+                query,
+                ["issue"],
+                expectedQuery1,
+                response1,
+                expectedQuery2,
+                response2,
+                hooks,
+                Mock(ResultComplexityAggregator)
+        )
+        then:
+        response == overallResponse
+        errors.collect { it.message } == errorMessages
+
+        where:
+        userId        || expectedQuery2                                                                     | overallResponse                                                                | errorMessages
+        "USER-SECRET" || 'query nadel_2_UserService {userQuery {userById(id:"USER-SECRET") {id}}}'          | [issue: [id: "ISSUE-1", author: [id: "USER-SECRET", restricted: null]]]        | ["removed field"]
+        "USER-OPEN"   || 'query nadel_2_UserService {userQuery {userById(id:"USER-OPEN") {id restricted}}}' | [issue: [id: "ISSUE-1", author: [id: "USER-OPEN", restricted: "superSecret"]]] | []
+    }
+
+    def "restrict fields based on hydration arguments with variables"() {
+        def overallSchema = TestUtil.schemaFromNdsl([
+                Issues     : '''
+                    service Issues {
+                        type Query {
+                            issue(id: ID): Issue
+                        }
+                        type Issue {
+                            id: ID
+                            author(productId: ID): User => hydrated from UserService.userById(id: $source.authorId, productId: $argument.productId) object identified by id
+                        }
+                    }
+        ''',
+                UserService: '''
+                    service UserService {
+                        type Query {
+                            userById(id: ID, productId: ID): User
+                        }
+                        type User {
+                            id: ID
+                            restricted: String
+                        }
+                    }
+        '''])
+        def issueSchema = TestUtil.schema("""
+        type Query {
+            issue(id: ID) : Issue
+        }
+        type Issue {
+            id: ID
+            authorId: ID
+               
+        }
+        """)
+        def userServiceSchema = TestUtil.schema("""
+        type Query {
+            userById(id: ID): User
+        }
+        type User {
+            id: ID
+            restricted: String
+        }
+        """)
+        def query = 'query issueQ($productId: ID) {issue(id: "ISSUE-1") { id author(productId: $productId) { id restricted }} }'
+        given:
+        def hooks = new ServiceExecutionHooks() {
+            @Override
+            CompletableFuture<Optional<GraphQLError>> isFieldForbidden(NormalizedQueryField normalizedField, HydrationArguments executionParameters, Object userSuppliedContext) {
```

Comment:
`HydrationArguments` is named `executionParameters`? that looks weird.

Reply:
also: I would not expect that you access the variables directly: I think you are only interested in argument values. So there should be a `getArgumentValue` method on `HydrationArguments`